test/hcl.go

```diff
@@ -93,6 +94,43 @@ func AssertDiff(t *testing.T, got, want interface{}, msg ...interface{}) {
 	}
 }
 
+func assertAssertsBlock(t *testing.T, got, want []hcl.AssertConfig) {
+	t.Helper()
+
+	if len(got) != len(want) {
+		t.Fatalf("got %d assert blocks, want %d", len(got), len(want))
+	}
+
+	for i, g := range got {
+		w := want[i]
+		assert.EqualStrings(t, w.Origin, g.Origin, "origin mismatch")
+		assert.EqualStrings(t,
+			exprAsStr(t, w.Assertion), exprAsStr(t, g.Assertion),
+			"assertion expr mismatch")
+		assert.EqualStrings(t,
+			exprAsStr(t, w.Message), exprAsStr(t, g.Message),
+			"message expr mismatch")
+		assert.EqualStrings(t,
+			exprAsStr(t, w.Warning), exprAsStr(t, g.Warning),
+			"warning expr mismatch")
+	}
+}
+
+// exprAsStr gets the original expression as a string for an expression built
+// with our
+func exprAsStr(t *testing.T, expr hhcl.Expression) string {
+	t.Helper()
+
+	if expr == nil {
```

Comment:
Not sure this is a good idea, we rarely test anything with empty expressions so it seemed safe to do this in order to avoid checking if both are nil before comparing etc.